### PR TITLE
Remove code from Workspaces.VB/C#.Desktop

### PR DESCRIPTION
**Pull request by jaredpar**
This change empties the MS.CA.Workspaces.CSharp.Desktop and MS.CA.Workspaces.VisualBasic.Desktop projects of all of their code.  The non-MSBuild specific logic was redistributed to the portable language specific workspace projects and the MSBuild specific logic was centralized into MS.CA.Workspaces.Desktop. 

This changed the strategy for building a `Workspace` object from an MSBuild file silghtly:

- The MSBuild parsing logic now builds up an `HostBuildOptions` object which is essentially a property bag of build data.
- Each language exports an `IHostBuildDataFactory` type which converts `HostBuildOptions` into a `HostBuildData` object which is just a tuple of `CompilationOptions` and `ParseOptions`.  
- The `HostBuildData` object is used to create the `ProjectFileInfo` type inside of `MSBuildWorkspaces`

### CODE REVIEW

**Comment by jaredpar** [comment]
@jasonmalinowski, @mattwar, @Pilchie, @tmat, @agocke

File: src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 namespace Microsoft.CodeAnalysis.Scripting
 {
     /// <summary>
-    /// Extends <see cref="MetadataFileReferenceResolver"/> to enable resolution of assembly
+    /// Extends MetadataFileReferenceResolver to enable resolution of assembly
```
Comment (jasonmalinowski) [inline src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs]: ...huh? It's the base type, so clearly it can be seen...?

**Comment by Pilchie** [comment]
LGTM

File: src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs]: @jasonmalinowski it's "by strange design".  The `cref` code doesn't care about `internal` accessibility and hence has an ambiguity between the accessible and inaccessible `MetadataFileReferenceResolver`.

Comment (jasonmalinowski) [inline src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs]: Yeah, you should complain to the compiler lead to get that fixed, where in the case of an ambiguity like this prefer the one in the assembly vs. one external to it.

**Comment by mattwar** [comment]
:+1:

File: src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs]: @jasonmalinowski I talked at length with the compiler lead and he just ignored me.

**Pull request merged**